regulations/tasks.py

```diff
@@ -35,47 +37,54 @@ def submit_comment(self, body, files):
-                ('uploadedFile', ('comment.pdf', comment)),
-                ('general_comment', 'See attached comment.pdf'),
-            ]
-
-            # Add other submitted fields
-            fields.extend([
-                (name, value)
-                for name, value in six.iteritems(body)
-                if name != 'general_comment'
-            ])
-            fields.extend(attachments)
-
-            data = MultipartEncoder(fields)
-            response = requests.post(
-                settings.REGS_GOV_API_URL,
-                data=data,
-                headers={
-                    'Content-Type': data.content_type,
-                    'X-Api-Key': settings.REGS_GOV_API_KEY,
-                }
-            )
-            if response.status_code != requests.codes.created:
-                logger.warn("Post to regulations.gov failed: %s %s",
-                            response.status_code, response.text)
-                raise self.retry()
-            logger.info(response.text)
-            return response.json()
-    except (ClientError, RequestException) as exc:
-        logger.exception(exc)
-        raise self.retry(exc=exc)
+        html = json_to_html(body)
+        try:
+            with html_to_pdf(html) as comment, \
+                    build_attachments(files) as attachments:
+                fields = [
+                    ('comment_on', settings.COMMENT_DOCUMENT_ID),
+                    # TODO: Ensure this name is unique
+                    ('uploadedFile', ('comment.pdf', comment)),
+                    ('general_comment', 'See attached comment.pdf'),
+                ]
+
+                # Add other submitted fields
+                fields.extend([
+                    (name, value)
+                    for name, value in six.iteritems(body)
+                    if name != 'general_comment'
+                ])
+                fields.extend(attachments)
+
+                data = MultipartEncoder(fields)
+                response = requests.post(
+                    settings.REGS_GOV_API_URL,
+                    data=data,
+                    headers={
+                        'Content-Type': data.content_type,
+                        'X-Api-Key': settings.REGS_GOV_API_KEY,
+                    }
+                )
+                if response.status_code != requests.codes.created:
+                    logger.warn("Post to regulations.gov failed: %s %s",
+                                response.status_code, response.text)
+                    raise self.retry()
+                logger.info(response.text)
+                return response.json()
+        except (RequestException, ClientError):
+            logger.exception("Unable to submit comment")
+            raise self.retry()
+    except MaxRetriesExceededError:
+        message = "Exceeded retries, saving failed submission"
+        logger.error(message)
+        failed_submission = FailedCommentSubmission(body=body, files=files)
```

Comment:
I originally had just the `body` in the DB schema for the very reason you point out. I decided with the _redundant_ `files` to make it easy to leverage `regulations.tasks.submit_comment` and write a driver for it from the DB, if and when we are pushed to do it.

Good point about not relying on Django for the stringification. I'll fix it. I'll also add type hints to the doc string to make it clear.